backend/TraderX/src/main/java/cmpe451/group6/rest/equipment/service/EquipmentUpdateService.java

```diff
@@ -0,0 +1,258 @@
+
+            @SuppressWarnings("unchecked")
+            Map<String, Map<String, String>> history = (Map<String, Map<String, String>>)
+                    jsonObject.toMap().get(CurrencyHistoryDTO.history);
+
+            @SuppressWarnings("unchecked")
+            Map<String, String> metadata = (Map<String, String>)
+                    jsonObject.toMap().get(CurrencyHistoryDTO.metadata);
+
+            if(history == null || metadata == null) {
+                System.err.println("Null response for currency history:" + currency);
+                continue;
+            }
+
+            saveSingleEquipmentHistory(history,metadata);
+        }
+
+        System.out.println("Currency history has been loaded.");
+
+    }
+
+    private void saveSingleEquipmentHistory(Map<String, Map<String, String>> history, Map<String, String> metadata){
+
+        String to = metadata.get(CurrencyHistoryDTO.toSymbol);
+
+        Equipment equipment = equipmentRepsitory.findByCode(to);
+
+        if(equipment == null) return;
+
+        equipment.getValueHistory().clear();
+
+        for (Map.Entry<String, Map<String, String>> daily : history.entrySet()) {
+            Date current;
+            try {
+                current = df.parse(daily.getKey());
+            } catch (ParseException e) {
+                e.printStackTrace();
+                continue;
+            }
+
+            double low = Double.parseDouble(daily.getValue().get(CurrencyHistoryDTO.low));
+            double high = Double.parseDouble(daily.getValue().get(CurrencyHistoryDTO.high));
+            double open = Double.parseDouble(daily.getValue().get(CurrencyHistoryDTO.open));
+            double close = Double.parseDouble(daily.getValue().get(CurrencyHistoryDTO.close));
+
+            Equipment.HistoricalValue hw = new Equipment.HistoricalValue(current,low,open,high,close);
+            equipment.addHistoricalValue(hw);
+
+        }
+
+        equipment.getValueHistory().sort((o1, o2) -> o1.getTimestamp().compareTo(o2.getTimestamp()));
+
+        equipmentRepsitory.save(equipment);
+
+    }
+
+
+    void updateEquipments(){
+
+        // Fetch real time values currently
+        // Store them in the DB.
+        // Do that every 9 A.M and 6 P.M.
+
+    }
+
+    void updateEquipmentsHistory(){
+
+        // Save last day's data to history.
+
+    }
+
+
+    double getNextPredictionValue(List<Equipment.HistoricalValue> history){
+
+        // Generate new prediction value from the last historical values.
+        return 0;
+    }
+
+
+    void saveEquipmentInfo(FileOutputStream os, String equipmentName){
+
+        // Persist equipment info on file.
+        // Might be called by a shutdown hook to prevent losing data between deployments.
+
+    }
+
+    void loadEquipmentInfo(FileInputStream is, String equipmentName){
+
+        // Load saved equipment info.
+        // Might be called on start up to prevent losing data between deployments.
+
+    }
+
+
+
```

Comment:
Just as these lines, too many empty lines exist

Reply:
You might be right. However, these were not the concern of mine when implementing service. Anyway, they're ignored in byte codes and IMO does not affect readability.